socorro/collector/wsgi_breakpad_collector.py

```diff
@@ -86,6 +91,33 @@ def POST(self, *args):
         else:
             raw_crash.legacy_processing = int(raw_crash.legacy_processing)
 
+        # We want to capture the crash report size, but need to differentiate
+        # between compressed vs. uncompressed data as well as accepted vs.
+        # rejected data.
+        crash_report_size = self._get_content_length()
+        is_compressed = self._is_content_gzipped()
+        is_accepted = (raw_crash.legacy_processing in (ACCEPT, DEFER))
+
+        try:
+            metrics_data = {}
+            size_key = '_'.join([
+                'crash_report_size',
+                'accepted' if is_accepted else 'rejected',
+                'compressed' if is_compressed else 'uncompressed',
+            ])
+            metrics_data = {
+                size_key: crash_report_size
+            }
+            self.metrics.capture_stats(metrics_data)
+        except Exception as exc:
+            # We *never* want metrics reporting to prevent saving a crash, so
+            # we catch everything and log an error.
+            self.logger.error(
+                'metrics kicked up exception: %s',
+                str(exc),
+                exc_info=True
+            )
```

Comment:
I'm pretty sure this section is safe. Does it look safe to you? If it goes sideways, does it tell us enough information that we can do something useful?